### PR TITLE
Add Youtube workload

### PR DESCRIPTION
**Pull request by muendelezaji**

### CODE REVIEW

File: wlauto/workloads/youtube/__init__.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    instrumentation_log = name + '_instrumentation.log'
+
+    def __init__(self, device, **kwargs):
+        super(Youtube, self).__init__(device, **kwargs)
+        self.run_timeout = 300
+        self.output_file = os.path.join(self.device.working_directory, self.instrumentation_log)
+
+    def validate(self):
+        super(Youtube, self).validate()
+        self.uiauto_params['package'] = self.package
```
Comment (ep1cman) [inline wlauto/workloads/youtube/__init__.py]: `validate` is meant just for validating parameters, setting of `uiauto_params` should be done in `initialize`

File: wlauto/workloads/youtube/__init__.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+import os
+import logging
+import re
+import time
+
+from wlauto import AndroidUiAutoBenchmark, Parameter
+from wlauto.exceptions import WorkloadError
+
+__version__ = '0.1.1'
```
Comment (ep1cman) [inline wlauto/workloads/youtube/__init__.py]: What is this for? Did we not discuss these before?

Comment (muendelezaji) [inline wlauto/workloads/youtube/__init__.py]: Leftover, removed.

File: wlauto/workloads/youtube/__init__.py (hunk 1 above)
Comment (muendelezaji) [inline wlauto/workloads/youtube/__init__.py]: This is moving to the superclass now

**Comment by muendelezaji** [comment]
Updated workload with `AndroidUxPerfWorkload` superclass and some adjustments for minor issues found while re-testing.

File: wlauto/workloads/youtube/__init__.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                  '''),
+    ]
+
+    view = [
+        package + '/com.google.android.apps.youtube.app.WatchWhileActivity',
+        package + '/com.google.android.apps.youtube.app.honeycomb.SettingsActivity',
+    ]
+
+    def __init__(self, device, **kwargs):
+        super(Youtube, self).__init__(device, **kwargs)
```
Comment (ep1cman) [inline wlauto/workloads/youtube/__init__.py]: `2016-08-31 10:34:25,653 CRITICAL root: TypeError: __init__() takes exactly 1 argument (13 given)`

Comment (muendelezaji) [inline wlauto/workloads/youtube/__init__.py]: Are you on the latest master? 

There was a problem with earlier superclass definition where `__init__` was defined as `def __init__(self)` i.e. no `device` and `**kwargs`.

But that was fixed in the merged commit. See https://github.com/ARM-software/workload-automation/commit/530714c61cefb5323890029823731a21d7fceca3#diff-78d101373d86c8b2294558ca4ae78c67R514

File: wlauto/workloads/youtube/__init__.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        self.run_timeout = 300
+
+    def validate(self):
+        super(Youtube, self).validate()
```
Comment (ep1cman) [inline wlauto/workloads/youtube/__init__.py]: this is not needed, WA does some meta magic on certain methods and automatically ensures their super is called. Validate is one of such methods:

https://github.com/ARM-software/workload-automation/blob/master/wlauto/core/extension.py#L404

File: wlauto/workloads/youtube/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    public static final String CLASS_VIEW_GROUP = "android.view.ViewGroup";
+
+    public static final int WAIT_TIMEOUT_1SEC = 1000;
+    public static final int WAIT_TIMEOUT_5SEC = 5000;
+    public static final int VIDEO_SLEEP_SECONDS = 3;
+    public static final int LIST_SWIPE_COUNT = 5;
+    public static final String SOURCE_MY_VIDEOS = "my_videos";
+    public static final String SOURCE_SEARCH = "search";
+    public static final String SOURCE_TRENDING = "trending";
+    public static final String[] STREAM_QUALITY = {
```
Comment (ep1cman) [inline wlauto/workloads/youtube/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java]: This isnt used

Comment (muendelezaji) [inline wlauto/workloads/youtube/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java]: Removed. Cleaned up other unused constants as well.

**Pull request merged**